### ISSUE TITLE
Enabling message tracking breaks mail delivery when using exim's sendmail

### ISSUE THREAD

**Issue by dhke** [sympa-community/sympa#1555]
When 

* message tracking is enabled for a list under "Bounce Management" and
* when `sendmail` is provided by exim

not messages get delivered to list subscribers. For a user and a list manager, this error is silent.

It would be nice if message tracking worked with exim or there was an option to disable it from the list management interface.

Version
-------
6.2.66

Installation method
-------------------
Ubuntu deb

Expected behavior
-----------------
Mails to list subscribers should be delivered with message tracking enabled.

Actual behavior
---------------
Mails get submitted to exim-sendmail, but not delivered because of command line options.

Steps to reproduce
------------------
1) Have a machine with `sendmail` provided by exim
2) Configure message tracking for a mailing list
3) Send mail to mailing list, watch the sendmail gets called (from logs)
4) See that messages get suppressed

Additional information
----------------------

The issue is in [Sympa/Mailer.pm](https://github.com/sympa-community/sympa/blob/95ec02cf7dbb55da09ae0c328f05b9f915bd6ec9/src/lib/Sympa/Mailer.pm#L107).

When message tracking is enabled, sendmail gets passed the command line flags `-N success,delay,failure -V  <envid>`

For postfix-sendmail, this sets an envelope id for DSN tracking and enables delivery status notifications for successful deliveries instead of the usual `delay, failure`.

For exim 

* `-N` is a [debugging option](https://www.exim.org/exim-html-current/doc/html/spec_html/ch-the_exim_command_line.html#vi86), that suppresses mail delivery. 
* `-V` makes exim print version information.

Setting message tracking for a list thus essentially disables mail delivery when using exim as the sendmail provider and there does not seem to be an option to disable the feature globally so that it is not available to list owners.

**Comment by ikedas** [comment]
You are right.  Currently the tracking feature supports Sendmail and Postfix.  The `sendmail` utilities bundled in the others, Exim, OpenSMTPD and probably more, do not provide options to enable tracking.

**Comment by ikedas** [comment]
So this is not a bug of Sympa.
The [documentation](https://www.sympa.community/manual/customize/message-tracking.html) doesn't say anything about a requirement for Sendmail or Postfix.  After adding such a note, I'd like to close this issue.  @dhke , do you agree?

**Comment by ikedas** [comment]
One more thing I forgot to mention.

If your MTA is Exim, I think you can enable tracking by setting the `dsn_advertise_hosts` option and using [smtpc](https://github.com/ikedas/smtpc) instead of `sendmail`.

**Comment by dhke** [comment]
> So this is not a bug of Sympa. The [documentation](https://www.sympa.community/manual/customize/message-tracking.html) doesn't say anything about a requirement for Sendmail or Postfix. After adding such a note, I'd like to close this issue. @dhke , do you agree?

That would be nice. Currently it's a silent breakage and one that can be accidentally triggered by a list owner.

If I understand this correctly, setting 

```tracking  privileged_owner  hidden```

in [edit_list.conf](https://github.com/sympa-community/sympa/tree/main/default/edit_list.conf) is probably a suggestion for those using exim.
